django/contrib/admin/tests.py

```diff
@@ -26,26 +56,35 @@ class AdminSeleniumWebDriverTestCase(StaticLiveServerTestCase):
         'django.contrib.sessions',
         'django.contrib.sites',
     ]
-    webdriver_class = 'selenium.webdriver.firefox.webdriver.WebDriver'
 
     @classmethod
     def setUpClass(cls):
-        if not os.environ.get('DJANGO_SELENIUM_TESTS', False):
-            raise SkipTest('Selenium tests not requested')
-        try:
-            cls.selenium = import_string(cls.webdriver_class)()
-        except Exception as e:
-            raise SkipTest('Selenium webdriver "%s" not installed or not '
-                           'operational: %s' % (cls.webdriver_class, str(e)))
-        cls.selenium.implicitly_wait(10)
+        '''
+        Validate the browsers requested and raise appropriate exception.
+        '''
+        browsers = os.environ.get('DJANGO_SELENIUM_SPECS', '').split(',')
```

Comment:
I'd do this validation in `runtests.py` -- no need to repeat it for every test classs.